### PR TITLE
LOG-3239: Add e2e test for vector http

### PR DESCRIPTION
**Pull request by vimalk78**
### Description
This PR adds a `e2e` test which sends logs from vector to a fluentd endpoint using http.

/cc 
/assign 

/cherry-pick 

### Links

- Depending on PR(s):
- Bugzilla:
- Github issue:
- JIRA:
- Enhancement proposal:

### CODE REVIEW

**Review by alanconway — APPROVED** [review]
/lgtm
is there a vector e2e test?

File: internal/generator/vector/output/http/http.go [hunk 1]
```diff
@@ -184,14 +185,19 @@ func Headers(o logging.OutputSpec) Element {
...
+		sortedKeys = append(sortedKeys, k)
+	}
+	sort.Strings(sortedKeys)
 	str := ""
 	skipComma := true
-	for k, v := range h {
+	for _, k := range sortedKeys {
 		if skipComma {
-			str += fmt.Sprintf("%q=%q", k, v)
+			str += fmt.Sprintf("%q=%q", k, h[k])
```
Comment (jcantrill) [inline internal/generator/vector/output/http/http.go:197]: What about using `strings.Join` in multiple iterations so there is no need to "skip" the comma or not?

**Review by jcantrill — CHANGES_REQUESTED** [review]
one nit but looks good

/hold

File: internal/generator/vector/output/http/http.go (hunk 1 above)
Comment (vimalk78) [reply internal/generator/vector/output/http/http.go]: updated.

**Pull request merged**